refactored how DocummentationComments is saved into the json.

Ali-chakaroun opened the pull request:
This is done to adhere to c# json schema.
in c# json we have "DocumentationComments": {}
now in java json we have "DocumentationComments" : { }

GuusVink commented on analyzer-java/src/main/java/com/infosupport/ldoc/analyzerj/descriptions/EnumMemberDescription.java:
```diff
@@ -13,12 +13,8 @@ public record EnumMemberDescription(
     List<Description> arguments,
 
     @JsonProperty("DocumentationComments")
-    @JsonInclude(JsonInclude.Include.NON_EMPTY)
```
Maybe NON_NULL would be nicer: this would prevent the json from containing a literal 'null' as well as still displaying an empty value '{ }' when the comment description is set

GuusVink commented on analyzer-java/src/main/java/com/infosupport/ldoc/analyzerj/descriptions/FieldDescription.java:
```diff
@@ -17,7 +17,6 @@ public record FieldDescription(
     String initialValue,
 
     @JsonProperty("DocumentationComments")
-    @JsonInclude(JsonInclude.Include.NON_EMPTY)
```
Same as other comment

GuusVink commented on analyzer-java/src/main/java/com/infosupport/ldoc/analyzerj/descriptions/MethodDescription.java:
```diff
@@ -12,7 +12,6 @@ public record MethodDescription(
     @JsonProperty("ReturnType")
     String returnType,
     @JsonProperty("DocumentationComments")
-    @JsonInclude(Include.NON_EMPTY)
```
idem

GuusVink commented on analyzer-java/src/main/java/com/infosupport/ldoc/analyzerj/descriptions/TypeDescription.java:
```diff
@@ -25,7 +25,6 @@ public record TypeDescription(
     List<String> baseTypes,
 
     @JsonProperty("DocumentationComments")
-    @JsonInclude(Include.NON_EMPTY)
```
Idem

GuusVink reviewed (commented):
Walked through it all, seems suitable.
Only one minor note regarding the complete removal of the JsonInclude annotation (which could also be replaced by a JsonIclude.NON_NULL)

Ali-chakaroun commented:
thank you for the review. I reviewed how the comments are done in the c# version and realized that i looked over some logic it uses. i initially believed in the c# json that "DocumentationComments": {}  is the result of empty comments that were required to be added into the json in any case, this was contradicted with "DocumentationComments" not displaying in the fields/constructor/etc... fields (in the json). Now i believe that the reason the c# json contains many  "DocumentationComments": {} is the result of manually adding the /// doc (in c# which equavalent to /** */ javadoc) but having empty values inside. i reverted the change to how the json handles empty vaues for DocumentationComments  and modified it to also display "DocumentationComments": {} when /** */ is added but with empty values.

wandernauta commented on analyzer-java/src/main/java/com/infosupport/ldoc/analyzerj/AnalysisVisitor.java:
```diff
@@ -71,12 +71,16 @@ public class AnalysisVisitor extends GenericListVisitorAdapter<Description, Anal
   private final SymbolResolver resolver;
 
-  /** Construct an AnalysisVisitor that uses the given SymbolResolver to find the names of types. */
+  /**
+   * Construct an AnalysisVisitor that uses the given SymbolResolver to find the names of types.
+   */
```
Here and below: do these really need to change? The single-line Javadoc is explicitly allowed per the [Java style guide](https://google.github.io/styleguide/javaguide.html#s7.1.1-javadoc-multi-line).

Ali-chakaroun replied:
I assume this all happened after i reformatted the code, wierd that it formatted them like that i didnt notice.

wandernauta reviewed (commented):
Not much to add to the review by @GuusVink except that I think the Javadoc changes are unnecessary - otherwise this looks like a reasonable refactor.

wandernauta commented:
@Ali-chakaroun Are you still looking at this? Both Guus' suggestion (use `JsonInclude.NON_NULL` instead of `JsonInclude.NON_EMPTY`) and my suggestion (revert unneeded changes to doc comments) still seem to apply.

Ali-chakaroun commented:
ye, i forgot to push them.

wandernauta commented on analyzer-java/src/main/java/com/infosupport/ldoc/analyzerj/descriptions/TypeDescription.java:
```diff
@@ -52,7 +52,9 @@ public record TypeDescription(
     implements Description {
 
-  /** Builder class for constructing TypeDescriptions incrementally. */
+  /**
+   * Builder class for constructing TypeDescriptions incrementally.
+   */
```
I would say these changes are also unneeded, so would suggest reverting them.

The reason I keep asking about these is that I do these reviews line-by-line, so if it _looks_ like something has changed but actually it hasn't, that means reviews take longer than necessary.

Ali-chakaroun replied:
problem is i have to manually revert these as my code formatter likes adding new lines to the javadoc.

wandernauta replied:
If you use IDEA, you can make sure the following checkbox is **checked**:

File > Settings > Editor > Code Style > Java > JavaDoc > Other > "Do not wrap one line comments"

But in general, I do recommend going line-by-line over your own pull requests as well and making sure the changes you're submitting are the ones you intend to submit.

wandernauta reviewed (commented):
Thanks! My suggestion to revert unneeded changes to Javadoc still stands, see below. Other than that, this looks good.

wandernauta reviewed (approved):
Thanks, no further questions!
